### DISCUSSION TITLE
How to add submit button for SearchField

### DISCUSSION THREAD

**Discussion by rhulse** [adobe/react-spectrum#6495] [Q&A] [answered]
I have implemented SearchField to replace an existing form in an app, and I cannot work out how to add a submit button.

All the SearchField handlers work correctly and I can press enter to search. 

When wrapping inside Form, and adding a button, the Form onSubmit handler does not return the value in the search field.

I have found [one example](https://react-spectrum.adobe.com/react-aria/SearchField.html#validation) in the docs which has the Button and the SearchField in a Form, but there is no indication of how to handle the onSubmit Button event correctly. 

The [example in StoryBook](https://react-spectrum.adobe.com/react-aria-tailwind-starter/index.html?path=/docs/searchfield--docs) wraps the SearchField and Button in an element <Y>. Also no help. :-(

**Comment by snowystinger** [comment]
I think this will address your issue when it's released https://github.com/adobe/react-spectrum/pull/6404

**Comment by rhulse** [comment]
Thanks @snowystinger. I am not sure. I had my SearchForm wrapped like this (irrelevant bit omitted):

```javascript
import { Button, Form, Input, Label, SearchField } from 'react-aria-components';

 <Form  onSubmit={onFormSubmit}>
      <SearchField>
        <Label className="hidden">{label}</Label>
        <Input
          placeholder={placeholder}
        />
        <Button
          type="submit"
        >
          <span className="hidden">Search</span>
          <SearchIcon className="w-5 h-5" />
        </Button>
      </SearchField>
    </Form>
```
All that <Button> does is fire the onSubmit handler on the the `<Form>` and there is no data from the input available. An onSubmit handler on the SearchField works fine when you press enter, but is not fired by the `<Button>` (of course).

**Reply by snowystinger** [reply, marked as answer]
The button you've passed to the SearchField will act as a clear button. https://react-spectrum.adobe.com/react-aria/SearchField.html#anatomy
You could put the submit button outside the SearchField and style the two together to look like a single field?

**Comment by rhulse** [comment]
Thanks.

Putting an outline of the code here for others.

I have done this:

```javascript
<Form  onSubmit={onFormSubmit}>
  <SearchField>
    <Label className="hidden">{label}</Label>
    <Input
      placeholder={placeholder}
    />
    </SearchField>
    <Button
      type="submit"
    >
      <span className="hidden">Search</span>
      <SearchIcon className="w-5 h-5" />
   </Button>
</Form>
```
In that case I can attach a separate event handler to the Form, for the key press, and process the data that way. Interestingly the docs say the event is `(event: FormEvent<[HTMLFormElement]>) => void...` but is actually of type `SyntheticBaseEvent`.

I was then able to handle the event this way:

```javascript
 const onFormSubmit = (event: SyntheticBaseEvent) => {
    event.preventDefault();
    console.log(event.target[0].value);
  };
```
